sunpy/map/map_factory.py

```diff
@@ -197,23 +197,22 @@ def _parse_args(self, *args, **kwargs):
                     i += 1    # an extra increment to account for the data-header pairing
 
             # File name
-            elif (isinstance(arg, str) and
-                  os.path.isfile(os.path.expanduser(arg))):
-                path = os.path.expanduser(arg)
+            elif (isinstance(arg, str) and pathlib.Path(arg).expanduser().is_file()):
+                path = pathlib.Path(arg).expanduser()
                 pairs = self._read_file(path, **kwargs)
                 data_header_pairs += pairs
 
             # Directory
-            elif (isinstance(arg, str) and
-                  os.path.isdir(os.path.expanduser(arg))):
-                path = os.path.expanduser(arg)
-                files = [os.path.join(path, elem) for elem in os.listdir(path)]
+            elif (isinstance(arg, str) and pathlib.Path(arg).expanduser().is_dir()):
+                path = pathlib.Path(arg).expanduser()
+                files = sorted(list(path.glob('*')))
                 for afile in files:
                     data_header_pairs += self._read_file(afile, **kwargs)
 
             # Glob
             elif (isinstance(arg, str) and '*' in arg):
-                files = glob.glob(os.path.expanduser(arg))
+                path = pathlib.Path(arg).expanduser().parent
+                files = sorted(list(path.glob('*')))
```

Comment:
This doesn't look correct...  Will this return just the files specified by the wildcard pattern, or will this return every file in the directory?

Reply:
yeah the latter.

Reply:
Oof you're right. This is not what should be happening here...

Reply:
Should be fixed now.